src/mono/mono/eventpipe/ep-rt-mono-runtime-provider.c

```diff
@@ -1340,15 +1340,15 @@ ep_rt_mono_sample_profiler_write_sampling_event_for_threads (
 }
 
 static void
-method_enter (MonoProfiler *prof, MonoMethod *method, MonoProfilerCallContext *ctx)
+sample_current_thread_stack_trace ()
 {
 	MonoThreadInfo *thread_info = mono_thread_info_current ();
 	SampleProfileStackWalkData stack_walk_data;
 	SampleProfileStackWalkData *data= &stack_walk_data;
 	THREAD_INFO_TYPE adapter = { { 0 } };
 
 	data->thread_id = ep_rt_thread_id_t_to_uint64_t (mono_thread_info_get_tid (thread_info));
-	data->thread_ip = (uintptr_t)MONO_CONTEXT_GET_IP (&ctx->context);
+	data->thread_ip = 0;
```

Comment:
What can I try to capture here ?

Reply:
WASM Interp doesn't set `INTERP_EXIT_WITH_CTX` and so LMF->ctx is invalid. It would point to `exit_pinvoke` label in the `ves_pinvoke_method` at best.

for mini (->AOT?) there is `MONO_INIT_CONTEXT_FROM_FUNC` which sets `ctx->wasm_ip` but I think it points to `mono_thread_state_init` at best.

The debugging symbols for WASM probably don't have anything that IP could be mapped to when interpreting the stack trace ?

Zero doesn't prevent the stack trace from working in the VisualStudio AFAIK. 

Conclusion, this is OK.

@lateralusX thoughts ?

Reply:
This is for single threaded only, so in theory WASM only scenario, correct? For the stack-walk, the interesting thing is the context, if you pass NULL we have logic inside mono_walk_stack_with_ctx to get the function address of mono_walk_stack_with_ctx as well as current SP/BP. Don't think data->thread_ip is currently used, so either you could do the same thing as mono_walk_stack_with_ctx does, setting up a context with MONO_INIT_CONTEXT_FROM_FUNC and use that to setup thread_ip as well as context parameter to mono_walk_stack_with_ctx or set it to 0. Getting the context would probably be more future proof if thread_ip gets used in the future.